src/libcommonserver/io/iohelper.cpp

```diff
@@ -598,6 +603,20 @@ bool IoHelper::checkIfPathExists(const SyncPath &path, bool &exists, IoError &io
         ioError = IoError::Success;
         return true;
     }
+#ifdef _WIN32 // TODO: Remove this block when migrating the release process to Visual Studio 2022.
+    // Prior to Visual Studio 2022, std::filesystem::symlink_status would return a missleading InvalidArgument if the path is
+    // found but located on a FAT32 disk. If the file is not found, it works as expected. This behavior is fixed when compiling
+    // with VS2022, see https://developercommunity.visualstudio.com/t/std::filesystem::is_symlink-is-broken-on/1638272
+    if (ioError == IoError::InvalidArgument && !Utility::isNtfs(path)) {
```

Comment:
path.remove_filename() ?

Reply:
`isNtfs `only use `SyncPath::root_name()` to determine the filesystem, there is no need to remove the filename.
```
bool Utility::isNtfs(const SyncPath &dirPath) {
    auto it = rootFsTypeMap.find(dirPath.root_name().string());
    if (it == rootFsTypeMap.end()) {
        std::string fsType = Utility::fileSystemName(dirPath);
        auto val = rootFsTypeMap.insert({dirPath.root_name().string(), fsType == NTFS});
        if (!val.second) {
            // Failed to insert into map
            return false;
        }
        it = val.first;
    }
    return it->second;
}
```